File: backend/internal/model/vo/scan_report.go
package vo

import "antd-gin-admin-backend/internal/model/entity"

// ScanReportVO is the API view of a Scan Report.
type ScanReportVO struct {
	ReportCode string `json:"report_code"`
	JobCode    string `json:"job_code"`
	Version    int    `json:"version"`
	ObjectKey  string `json:"object_key"`
	Format     string `json:"format"`
	CreatedAt  string `json:"created_at"`
}

func BuildScanReportVO(r *entity.ScanReport) *ScanReportVO {
	if r == nil {
		return nil
	}
	return &ScanReportVO{
		ReportCode: r.ReportCode,
		JobCode:    r.JobCode,
		Version:    r.Version,
		ObjectKey:  r.ObjectKey,
		Format:     r.Format,
		CreatedAt:  r.CreatedAt.Format("2006-01-02 15:04:05"),
	}
}

File: frontend/admin-web/src/services/antd-gin-api/scan_report.ts
import { request } from '@umijs/max';
import type { PageResult } from './asset';

export interface ScanReport {
  report_code: string;
  job_code: string;
  version: number;
  object_key: string;
  format: string;
  created_at: string;
}

export interface ListScanReportParams {
  page?: number;
  pageSize?: number;
  job_code?: string;
}

export interface CreateScanReportRequest {
  job_code: string;
}

/** 获取扫描报告列表 */
export async function getScanReportList(params?: ListScanReportParams) {
  return request<{
    code: number;
    data: PageResult<ScanReport>;
    message?: string;
  }>('/api/v1/scan/scan-reports', {
    method: 'GET',
    params,
  });
}

/** 生成扫描报告 */
export async function createScanReport(data: CreateScanReportRequest) {
  return request<{
    code: number;
    data: ScanReport;
    message?: string;
  }>('/api/v1/scan/scan-reports', {
    method: 'POST',
    data,
  });
}

/** 下载扫描报告（HTML blob） */
export async function downloadScanReport(reportCode: string) {
  return request<Blob>(`/api/v1/scan/scan-reports/${reportCode}/download`, {
    method: 'GET',
    responseType: 'blob',
  });
}
